### PR TITLE
[Merged by Bors] - chore(Set/Finset): standardize names of distributivity laws

### PR DESCRIPTION
**Pull request by FordUniver**
Standardizes the following names for distributivity laws across `Finset` and `Set`:

- `inter_union_distrib_left`
- `inter_union_distrib_right`
- `union_inter_distrib_left`
- `union_inter_distrib_right`

Makes arguments explicit in:

- `Set.union_inter_distrib_right`
- `Set.union_inter_distrib_left`
- `Set.inter_union_distrib_right`

Deprecates these theorem names:

- `Finset.inter_distrib_left`
- `Finset.inter_distrib_right`
- `Finset.union_distrib_right`
- `Finset.union_distrib_left`
- `Set.inter_distrib_left`
- `Set.inter_distrib_right`
- `Set.union_distrib_right`
- `Set.union_distrib_left`

Fixes use of deprecated names and implicit arguments in these files:

- Topology/Basic
- Topology/Connected/Basic
- MeasureTheory/MeasurableSpace/Basic
- MeasureTheory/Covering/Differentiation
- MeasureTheory/Constructions/BorelSpace/Basic
- Data/Set/Image
- Data/Set/Basic
- Data/PFun
- Data/Matroid/Dual
- Data/Finset/Sups
- Data/Finset/Basic
- Combinatorics/SetFamily/FourFunctions
- Combinatorics/Additive/SalemSpencer
- Counterexamples/Phillips.lean
- Archive/Imo/Imo2021Q1.lean

### CODE REVIEW

File: Mathlib/Data/Finset/Basic.lean [hunk 1]
```diff
@@ -1803,21 +1803,45 @@ theorem inter_left_idem (s t : Finset α) : s ∩ (s ∩ t) = s ∩ t := inf_lef
 theorem inter_right_idem (s t : Finset α) : s ∩ t ∩ t = s ∩ t := inf_right_idem _ _
 #align finset.inter_right_idem Finset.inter_right_idem
 
+theorem inter_union_distrib_left (s t u : Finset α) : s ∩ (t ∪ u) = s ∩ t ∪ s ∩ u :=
+  inf_sup_left _ _ _
+#align finset.inter_union_distrib_left Finset.inter_union_distrib_left
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Basic.lean:1808]: Don't create `#align`s! They correspond to Lean 3 declarations and `finset.inter_union_distrib_left` never was a Lean 3 declaration.
```suggestion
#align finset.inter_distrib_left Finset.inter_union_distrib_left
```
and below

File: Mathlib/Data/Finset/Basic.lean [hunk 2]
```diff
@@ -1803,21 +1803,45 @@ theorem inter_left_idem (s t : Finset α) : s ∩ (s ∩ t) = s ∩ t := inf_lef
...
+
+/-- Deprecated, use `Finset.inter_union_distrib_left` instead. -/
+@[deprecated inter_union_distrib_left]
 theorem inter_distrib_left (s t u : Finset α) : s ∩ (t ∪ u) = s ∩ t ∪ s ∩ u :=
   inf_sup_left _ _ _
 #align finset.inter_distrib_left Finset.inter_distrib_left
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Basic.lean:1814]:
```suggestion
-- 2024-03-22
@[deprecated] alias inter_distrib_left := inter_union_distrib_left
```
and below

**Review by YaelDillies — COMMENTED** [review]
and in other places.

**Comment by YaelDillies** [comment]
You should add `-- 2024-03-22` before each (block of) `@[deprecated]` so that we know how old the deprecation is.

File: Mathlib/Data/Finset/Basic.lean [hunk 3]
```diff
@@ -1803,21 +1803,27 @@ theorem inter_left_idem (s t : Finset α) : s ∩ (s ∩ t) = s ∩ t := inf_lef
...
   sup_inf_left _ _ _
-#align finset.union_distrib_left Finset.union_distrib_left
+#align finset.union_distrib_left Finset.union_inter_distrib_left
 
-theorem union_distrib_right (s t u : Finset α) : s ∩ t ∪ u = (s ∪ u) ∩ (t ∪ u) :=
-  sup_inf_right _ _ _
-#align finset.union_distrib_right Finset.union_distrib_right
+theorem union_inter_distrib_right (s t u : Finset α) : (s ∪ t) ∩ u = s ∩ u ∪ t ∩ u :=
+  inf_sup_right _ _ _
+#align finset.inter_distrib_right Finset.union_inter_distrib_right
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Basic.lean:1820]: Can you keep those three the order they were before?

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**